src/main/kotlin/com/sourcegraph/cody/chat/ui/MessagesPanel.kt

```diff
@@ -54,7 +54,12 @@ class MessagesPanel(private val project: Project, private val chatSession: ChatS
 
   @RequiresEdt
   fun removeBlinkingCursor() {
-    components.find { it is BlinkingCursorComponent }?.let { remove(it) }
+    components
+        .firstNotNullOfOrNull { it as? BlinkingCursorComponent }
+        ?.let {
+          it.dispose()
```

Comment:
The Jetbrains documentation [says](https://plugins.jetbrains.com/docs/intellij/disposers.html#diagnosing-disposer-leaks) that `Disposable.dispose()` should never be called directly. However, it turns out that we do this _a lot_ in our source code, and I don't expect this PR to rectify all those occurrences.

Reply:
good point - I fixed it in this place. I will review the other places later.